aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::natives::{
+    aggregator_natives::helpers_v2::{
+        aggregator_snapshot_string_info, aggregator_snapshot_u128_info,
+        aggregator_snapshot_u64_info, string_value_info,
+    },
+    AccountAddress,
+};
+use aptos_gas_schedule::gas_params::natives::aptos_framework::*;
+use aptos_native_interface::{
+    safely_pop_arg, RawSafeNative, SafeNativeBuilder, SafeNativeContext, SafeNativeError,
+    SafeNativeResult,
+};
+use move_core_types::value::{MoveStructLayout, MoveTypeLayout};
+use move_vm_runtime::native_functions::NativeFunction;
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Struct, StructRef, Value},
+};
+use smallvec::{smallvec, SmallVec};
+use std::{collections::VecDeque, ops::Deref};
+
+/// The generic type supplied to aggregator snapshots is not supported
+pub const EUNSUPPORTED_AGGREGATOR_SNAPSHOT_TYPE: u64 = 0x02_0005;
+
+/// The aggregator snapshots feature is not enabled
+pub const EAGGREGATOR_SNAPSHOTS_NOT_ENABLED: u64 = 0x02_0006;
+
+/// Checks if the type argument `type_arg` is a string type
+fn is_string_type(context: &SafeNativeContext, type_arg: &Type) -> SafeNativeResult<bool> {
```

Comment:
mhm, should this function live in some place for utilities? in fact in a more general utility place (for / with types) than just aggregators?

Reply:
maybe then it can be applied to ty and we can call it after getting ty, and that can have a nice return type of bool.

`context.deref().type_to_fully_annotated_layout(type_arg).map(general_type_utility_place::is_string_type)`

at that point we may even be able to not have an additional method here and just do the above at current callsite.